independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationLiteralGenerator.java

```diff
@@ -252,8 +252,14 @@ static String componentType(MethodInfo method) {
     }
 
     static String generatedSharedName(DotName annotationName) {
+        // when the annotation is a java.lang annotation we need to use a different package in which to generate the literal
+        // otherwise a security exception will be thrown when the literal is loaded
+        String nameToUse = annotationName.toString().startsWith("java.lang")
```

Comment:
Looks good. Could we use the same logic for `generatedLocalName()` as well. It's not used for Quarkus but we should be consistent ;-).

Reply:
Makes perfect sense! Done :)